File: src/main/java/com/trade/securities/adapter/in/web/StockDataController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.LoadStockDataUseCase;
import com.trade.securities.domain.StockData;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/stock-data")
@RequiredArgsConstructor
public class StockDataController {

    private final LoadStockDataUseCase loadStockDataUseCase;

    @GetMapping
    public List<StockData> getStockData(
            @RequestParam String date,
            @RequestParam(defaultValue = "KOSDAQ") String type) {
        return loadStockDataUseCase.getStockData(date, type);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/StockDataController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.LoadStockDataUseCase
import com.trade.securities.domain.StockData
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.DefaultValue
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/stock-data")
@Produces(MediaType.APPLICATION_JSON)
class StockDataController(
    private val loadStockDataUseCase: LoadStockDataUseCase
) {

    @GET
    fun getStockData(
        @QueryParam("date") date: String,
        @QueryParam("type") @DefaultValue("KOSDAQ") type: String
    ): List<StockData> {
        return loadStockDataUseCase.getStockData(date, type)
    }
}
